studips/src/component/MainThread.js

```diff
@@ -65,8 +65,15 @@ class MainThread extends React.Component {
 	}
 	getTabContent() {
 		let posts = this.state.posts;
-		if (this.state.city.length > 0) {
-			posts = posts.filter(post => post.city.toLowerCase() === this.state.city.toLowerCase());
+		if (this.state.search.length > 0) {
+			posts = posts.filter(
+				post => (
+					post.city.toLowerCase() === this.state.search.toLowerCase() ||
+					post.firstname.toLowerCase() === this.state.search.toLowerCase()||
+					post.lastname.toLowerCase() === this.state.search.toLowerCase()||
+					post.study.toLowerCase() === this.state.search.toLowerCase()
+				)
```

Comment:
try:

    post => ([
        post.city,
        post.firstname,
        post.lastname,
        post.study,
    ].map(string => string.toLowerCase()).includes(this.state.search.toLowerCase()))